packages/block-library/src/post-tags/edit.js

```diff
@@ -1,42 +1,77 @@
 /**
  * WordPress dependencies
  */
-import { useEntityProp, useEntityId } from '@wordpress/core-data';
+import { useEntityProp } from '@wordpress/core-data';
+import { Warning, __experimentalBlock as Block } from '@wordpress/block-editor';
 import { useSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
-function PostTagsDisplay() {
-	const [ tags ] = useEntityProp( 'postType', 'post', 'tags' );
+export default function PostTagsEdit( { context } ) {
+	const [ tags ] = useEntityProp(
+		'postType',
+		context.postType,
+		'tags',
+		context.postId
+	);
 	const tagLinks = useSelect(
 		( select ) => {
 			const { getEntityRecord } = select( 'core' );
 			let loaded = true;
-			const links = tags.map( ( tagId ) => {
-				const tag = getEntityRecord( 'taxonomy', 'post_tag', tagId );
-				if ( ! tag ) {
-					return ( loaded = false );
-				}
-				return (
-					<a key={ tagId } href={ tag.link }>
-						{ tag.name }
-					</a>
-				);
-			} );
+			let links;
+
+			if ( tags ) {
+				links = tags.map( ( tagId ) => {
+					const tag = getEntityRecord(
+						'taxonomy',
+						'post_tag',
+						tagId
+					);
+					if ( ! tag ) {
+						return ( loaded = false );
+					}
+					return (
+						<a key={ tagId } href={ tag.link }>
+							{ tag.name }
+						</a>
+					);
+				} );
+			}
+
 			return loaded && links;
 		},
 		[ tags ]
 	);
-	return (
+
+	let display =
 		tagLinks &&
 		( tagLinks.length === 0
 			? __( 'No tags.' )
-			: tagLinks.reduce( ( prev, curr ) => [ prev, ' | ', curr ] ) )
-	);
-}
+			: tagLinks.reduce( ( prev, curr ) => [ prev, ' | ', curr ] ) );
 
-export default function PostTagsEdit() {
-	if ( ! useEntityId( 'postType', 'post' ) ) {
-		return __( 'Post Tags' );
+	if ( ! context.postType || ! context.postId ) {
+		display = (
+			<Warning>
+				{ __( 'Post tags block: No post found for this block.' ) }
+			</Warning>
+		);
+
+		/**
```

Comment:
I'd probably move the comment after `} else if ( context.postType !== 'post' ) {`.
It threw me a bit off guard while quickly scanning the code, because I was expecting to see the "target" of the comment _inside_ that block (likely because of the indentation).

Reply:
👍

Bernie and I had a [discussion](https://github.com/WordPress/gutenberg/pull/24082#discussion_r459720636) about indentation where I tried to remedy this (but couldn't). The formatting is unfortunate, given the [WP style guide's](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#comments) rule to keep "targets" below the comment, but this seems like a good place to make an exception.

Making the change now.